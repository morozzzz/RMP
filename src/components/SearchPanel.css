.search-pannel {
    display: flex;
    flex-direction: column;
    color: white;
    width: 100%;
}

.search-bottom-pannel{
    display: flex;
    margin: 1em 0;
    justify-content: space-between;
}

.search-criterias {
    display: flex;
}

.search-criterias {
    display: flex;
    width: 18em;
    justify-content: space-between;
    align-items: center;
}

.criteria-button {
    background-color: rgb(51, 51, 51);
    height: 2em;
    width: 6em;
    display: flex;
    justify-content: center;
    align-items: center;
    border: none;
    outline: none;
    color: white;
}

.criteria-button-selected {
    background-color: rgb(248, 129, 129);
}

.search-input {
    outline: none;
    background-color: black;
    border: none;
    border-bottom: 0.1em solid rgb(248, 129, 129);
    height: 3em;
    color: white;
}

.search-button {
    background-color: rgb(248, 129, 129);
    border: none;
    outline: none;
    color: white;
    width: 15em;
}

.search-button:disabled {
    background-color: rgba(248, 129, 129, 0.685);
}

.search-button:active {
    box-shadow: 0 0 0.5em white;
}

.pointer {
    cursor: pointer;
}