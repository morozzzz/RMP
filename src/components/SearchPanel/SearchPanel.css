.search-panel {
    display: flex;
    flex-direction: column;
    color: white;
    width: 100%;
}

.search-panel__bottom-bar{
    display: flex;
    margin: 1em 0;
    justify-content: space-between;
}

.search-panel__criterias {
    display: flex;
}

.search-panel__criterias {
    display: flex;
    width: 18em;
    justify-content: space-between;
    align-items: center;
}

.search-panel__criteria-button {
    background-color: rgb(51, 51, 51);
    height: 2em;
    width: 6em;
    display: flex;
    justify-content: center;
    align-items: center;
    border: none;
    outline: none;
    color: white;
    cursor: pointer;
}

.search-panel__criteria-button_selected {
    background-color: rgb(248, 129, 129);
}

.search-panel__input {
    outline: none;
    background-color: black;
    border: none;
    border-bottom: 0.1em solid rgb(248, 129, 129);
    height: 3em;
    color: white;
}

.search-panel__search-button {
    background-color: rgb(248, 129, 129);
    border: none;
    outline: none;
    color: white;
    width: 15em;
    cursor: pointer;
}

.search-panel__search-button:disabled {
    background-color: rgba(248, 129, 129, 0.685);
}

.search-panel__search-button:active {
    box-shadow: 0 0 0.5em white;
}
