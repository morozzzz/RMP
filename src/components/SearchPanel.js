import React from 'react';
import './SearchPanel.css';

const CRITERIAS = {
    TITLE: 'title',
    GENRE: 'genre',
};

class SearchPannel extends React.Component {
    constructor() {
        super();
        this.state = {
            inputData: null,
            searchCriteria: CRITERIAS.TITLE,
        };
    }

    onInputChange = (event) => {
        this.setState({ inputData: event.target && event.target.value });
    };

    onTitleCriteriaClick = () => {
        this.setState({ searchCriteria: CRITERIAS.TITLE });
    };

    onGenreCriteriaClick = () => {
        this.setState({ searchCriteria: CRITERIAS.GENRE });
    };

    onSearchClick = () => {
    };

    render() {
        const { inputData, searchCriteria } = this.state;

        return (
            <div className="search-pannel">
                <h3 className="search-header">FIND YOUR MOVIE</h3>
                <input className="search-input" type="text" placeholder="Enter key words..." onChange={this.onInputChange} />
                <div className="search-bottom-pannel">
                    <div className="search-criterias">
                        <div>SEARCH BY</div>
                        <button onClick={this.onTitleCriteriaClick} type="button" className={`criteria-button pointer ${searchCriteria === CRITERIAS.TITLE && 'criteria-button-selected'}`}>TITLE</button>
                        <button onClick={this.onGenreCriteriaClick} type="button" className={`criteria-button pointer ${searchCriteria === CRITERIAS.GENRE && 'criteria-button-selected'}`}>GENRE</button>
                    </div>
                    <button onClick={this.onSearchClick} disabled={!inputData} type="button" className="search-button pointer">SEARCH</button>
                </div>
            </div>
        );
    }
}

export default SearchPannel;
