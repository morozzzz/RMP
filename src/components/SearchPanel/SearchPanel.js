import React from 'react';
import './SearchPanel.css';

const CRITERIAS = {
    TITLE: 'title',
    GENRE: 'genre',
};

class SearchPannel extends React.Component {
    constructor() {
        super();
        this.state = {
            inputData: '',
            searchCriteria: CRITERIAS.TITLE,
        };
    }

    onInputChange = (event) => {
        this.setState({ inputData: event.target && event.target.value });
    };

    onTitleCriteriaClick = () => {
        this.setState({ searchCriteria: CRITERIAS.TITLE });
    };

    onGenreCriteriaClick = () => {
        this.setState({ searchCriteria: CRITERIAS.GENRE });
    };

    onSearchClick = () => {
        console.log(999);
    };

    handleKey = (event) => {
        const { inputData } = this.state;
        if (event.keyCode === 13 && inputData) {
            this.onSearchClick();
        }
    }

    render() {
        const { inputData, searchCriteria } = this.state;

        return (
            <div className="search-panel">
                <h3 className="search-panel__header">FIND YOUR MOVIE</h3>
                <input className="search-panel__input" type="text" placeholder="Enter key words..." onKeyUp={this.handleKey} onChange={this.onInputChange} />
                <div className="search-panel__bottom-bar">
                    <div className="search-panel__criterias">
                        <div>SEARCH BY</div>
                        <button onClick={this.onTitleCriteriaClick} type="button" className={`search-panel__criteria-button ${searchCriteria === CRITERIAS.TITLE && 'search-panel__criteria-button_selected'}`}>TITLE</button>
                        <button onClick={this.onGenreCriteriaClick} type="button" className={`search-panel__criteria-button ${searchCriteria === CRITERIAS.GENRE && 'search-panel__criteria-button_selected'}`}>GENRE</button>
                    </div>
                    <button onClick={this.onSearchClick} disabled={!inputData} type="button" className="search-panel__search-button">SEARCH</button>
                </div>
            </div>
        );
    }
}

export default SearchPannel;
