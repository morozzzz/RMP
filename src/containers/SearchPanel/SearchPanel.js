import React from 'react';
import { connect } from 'react-redux';
import { CRITERIAS, KEY_CODES } from '../../constants/app.constants';
import {
    setActiveCrtiteria,
    updateSearchText,
} from '../../actions/search.actions';

import './SearchPanel.css';

class SearchPannel extends React.Component {
    handleSearchClick = () => {
        const { onSearchClick } = this.props;

        onSearchClick();
    };

    handleKey = (event) => {
        const { onSearchClick, search } = this.props;

        if (event.keyCode === KEY_CODES.ENTER && search.trim()) {
            onSearchClick();
        }
    }

    render() {
        const {
            onCriteriaClick,
            search,
            searchBy,
            onInputChange,
        } = this.props;

        return (
            <div className="search-panel">
                <h3 className="search-panel__header">FIND YOUR MOVIE</h3>
                <input className="search-panel__input" type="text" placeholder="Enter key words..." onKeyUp={this.handleKey} onChange={onInputChange} />
                <div className="search-panel__bottom-bar">
                    <div className="search-panel__criterias">
                        <div>SEARCH BY</div>
                        <button id="title-button" value={CRITERIAS.TITLE} onClick={onCriteriaClick} type="button" className={`search-panel__criteria-button ${searchBy === CRITERIAS.TITLE && 'search-panel__criteria-button_selected'}`}>TITLE</button>
                        <button id="genre-button" value={CRITERIAS.GENRE} onClick={onCriteriaClick} type="button" className={`search-panel__criteria-button ${searchBy === CRITERIAS.GENRE && 'search-panel__criteria-button_selected'}`}>GENRE</button>
                    </div>
                    <button onClick={this.handleSearchClick} disabled={!search.trim()} type="button" className="search-panel__search-button">SEARCH</button>
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => (
    {
        search: state.searchParams.search,
        searchBy: state.searchParams.searchBy,
    }
);

const mapDispatchToProps = dispatch => (
    {
        onCriteriaClick: event => dispatch(setActiveCrtiteria(event.target.value)),
        onInputChange: event => dispatch(updateSearchText(event.target.value)),
    }
);
export default connect(mapStateToProps, mapDispatchToProps)(SearchPannel);
