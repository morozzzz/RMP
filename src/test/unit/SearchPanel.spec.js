import React from 'react';
import { shallow } from 'enzyme';
import SearchPanel from '../../containers/SearchPanel/SearchPanel';

describe('SearchPanel', () => {
    it('should be rendered correctly', () => {
        const wrapper = shallow(<SearchPanel />);

        expect(wrapper).toMatchSnapshot();
    });
});
