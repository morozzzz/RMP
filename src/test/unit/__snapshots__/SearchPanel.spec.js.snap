// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel should be rendered correctly 1`] = `
<div
  className="search-panel"
>
  <h3
    className="search-panel__header"
  >
    FIND YOUR MOVIE
  </h3>
  <input
    className="search-panel__input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Enter key words..."
    type="text"
  />
  <div
    className="search-panel__bottom-bar"
  >
    <div
      className="search-panel__criterias"
    >
      <div>
        SEARCH BY
      </div>
      <button
        className="search-panel__criteria-button search-panel__criteria-button_selected"
        id="title-button"
        onClick={[Function]}
        type="button"
      >
        TITLE
      </button>
      <button
        className="search-panel__criteria-button false"
        id="genre-button"
        onClick={[Function]}
        type="button"
      >
        GENRE
      </button>
    </div>
    <button
      className="search-panel__search-button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      SEARCH
    </button>
  </div>
</div>
`;
